org.sonatype.m2e.egit/src/org/sonatype/m2e/egit/internal/EgitScmHandler.java

```diff
@@ -76,9 +99,30 @@ protected int getTimeout() {
     return 30;
   }
 
-  protected URIish getUri(MavenProjectScmInfo info) throws URISyntaxException {
+  protected void runCloneOperation(URIish uri, File location, String refName, SubMonitor pm)
+                  throws InvocationTargetException, IOException, InterruptedException {
+    CloneOperation clone = new CloneOperation(uri, true /* allSelected */, new ArrayList<Ref>(), location, refName,
+          "origin", getTimeout());
+    clone.run(pm.newChild(99));
+
+    fixAutoCRLF(clone.getGitDir());
+  }
+
+  protected boolean onAuthFailed() {
+    final boolean[] result = new boolean[]{false};
+    Display.getDefault().syncExec(new Runnable() {
+      @Override
+      public void run() {
+        Shell shell = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();
+        result[0] = MessageDialog.openConfirm(shell, "Auth failed", "URL has SSH protocol. It seems you do not have a valid SSH key. Do you want to continue with GIT protocol anonymously?");
```

Comment:
Authentication failed
The clone URL uses the SSH protocol. It seems you do not have a valid SSH key. Do you want to continue with the GIT protocol anonymously?